### PR TITLE
Feature/mat 4076

### PR DESCRIPTION
**Pull request by RohitKandimalla**
## CQL to ELM Translation Service PR

Jira Ticket: [MAT-4076](https://jira.cms.gov/browse/MAT-4076)
(Optional) Related Tickets:

### Summary

- Added okta security to the service.
- Calling the madie-fhir-services using the accessToken captured from the request. 
- Enabled CORS.
- Added logInterceptor.

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/mat/cql_elm_translation/service/MadieFhirServices.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+@RequiredArgsConstructor
+public class MadieFhirServices {
+
+    private final RestTemplate restTemplate;
+
+    @Value("${madie.fhir.service.baseUrl}")
+    private String madieFhirService;
+
+    @Value("${madie.fhir.service.libraries.uri}")
+    private String librariesUri;
```
Comment (adongare) [inline src/main/java/gov/cms/mat/cql_elm_translation/service/MadieFhirServices.java:27]: Could this be a little more explicit?

**Review by adongare — APPROVED** [review]

File: src/main/java/gov/cms/mat/cql_elm_translation/service/MadieFhirServices.java (hunk 1 above)
Comment (RohitKandimalla) [reply src/main/java/gov/cms/mat/cql_elm_translation/service/MadieFhirServices.java:27]: its a long name, but I updated it by adding hapi-fhir to it. Please take a look. Thank you

**Review by adongare — APPROVED** [review]
